source/common/http/conn_manager_impl.cc

```diff
@@ -143,15 +145,15 @@ void ConnectionManagerImpl::initializeReadFilterCallbacks(Network::ReadFilterCal
   }
 
   if (config_.idleTimeout()) {
-    connection_idle_timer_ = read_callbacks_->connection().dispatcher().createScaledTimer(
-        Event::ScaledTimerType::HttpDownstreamIdleConnectionTimeout,
-        [this]() -> void { onIdleTimeout(); });
+    connection_idle_timer_ =
+        dispatcher_->createScaledTimer(Event::ScaledTimerType::HttpDownstreamIdleConnectionTimeout,
+                                       [this]() -> void { onIdleTimeout(); });
```

Comment:
A very simple replacement from `read_callbacks_->connection().dispatcher()` to single `dispatcher_`. In this class, there are 19 positions need use reference of dispatcher and `read_callbacks_->connection().dispatcher()` is called repeatly in 19 positions.

Reply:
Can you split this out into a different PR? There is some memory overhead from this change. Not sure how that compares to (maybe) some CPU savings depending on compiler optimizations.

/wait

Reply:
got it.